qiskit_experiments/curve_analysis/curve_analysis.py

```diff
@@ -240,14 +240,15 @@ def __init__(self):
         """Initialize data fields that are privately accessed by methods."""
         super().__init__()
 
-        #: Dict[str, Any]: Experiment metadata
-        self.__experiment_metadata = None
-
         #: List[CurveData]: Processed experiment data set.
         self.__processed_data_set = list()
 
-        #: Backend: backend object used for experimentation
-        self.__backend = None
+        # Number of qubits and backend for curve analysis
+        # TODO: these should be removed as attributes in this class,
+        # however they are currently required RBAnalysis and RBUtils
+        # which needs refactoring to remove this dependency.
+        self._num_qubits = None
+        self._backend = None
```

Comment:
This should be okey but we can first update RB to remove backend dependency. This is used to extract error ratio of single qubit gates, but logic is not correct because it should assume a particular situation. e.g.
- basis_gates: sx, rz, x ... sx and x are almost same error ignoring nonlinearity 
- basis_gates: u1, u2, u3 ... u3:u2 = 2:1 by assuming specific implementation with different numbers of sx gate, rz=error free
- basis_gates: sx, sy, rz ... sx and sy are the same error, e.g. rz-sx-rz^dag

If we have two gates consisting of different pulse sequence in the basis gate this doesn't work. So I think this error ratio dictionaly should be provided without extracting something from backend.